controllers/lease_controller.go

```diff
@@ -0,0 +1,456 @@
+					Annotations: etcd.Spec.Annotations,
+					Labels:      getLabels(etcd),
+				},
+				Spec: v1.PodSpec{
+					//ServiceAccountName: fmt.Sprint(values["serviceAccountName"]),
+					RestartPolicy: v1.RestartPolicyNever,
+					Containers: []v1.Container{{
+						Name:            "compact-backup",
+						Image:           *etcd.Spec.Backup.Image,
+						ImagePullPolicy: v1.PullIfNotPresent,
+						Command:         getCompactJobCommands(etcd),
+						VolumeMounts:    getCmpctJobVolumeMounts(etcd),
+						Env:             getCmpctJobEnvVar(etcd),
+					}},
+					Volumes: getCmpctJobVolumes(etcd),
+				},
+			},
+		},
+	}
+
+	if etcd.Spec.Backup.CompactionResources != nil {
+		job.Spec.Template.Spec.Containers[0].Resources = *etcd.Spec.Backup.CompactionResources
+	}
+
+	logger.Info("Creating job", "job", kutil.Key(job.Namespace, job.Name).String())
+	err = lc.Create(ctx, job)
+
+	// Ignore the precondition violated error, this machine is already updated
+	// with the desired label.
+	if err == errorsutil.ErrPreconditionViolated {
+		logger.Info("Job precondition doesn't hold, skip updating it", "job", kutil.Key(job.Namespace, job.Name).String())
+		err = nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	//TODO (abdasgupta): Evaluate necessity of claiming object here after creation
+	return job, nil
+}
+
+func getJobName(etcd *druidv1alpha1.Etcd) string {
+	return fmt.Sprintf("%s-compact-backup", string(etcd.UID[:6]))
+}
+
+func getLabels(etcd *druidv1alpha1.Etcd) map[string]string {
+	return map[string]string{
+		"name":     "etcd",
+		"instance": etcd.Name,
+	}
+}
+func getCmpctJobVolumeMounts(etcd *druidv1alpha1.Etcd) []v1.VolumeMount {
+	vms := []v1.VolumeMount{
+		{
+			Name:      "etcd-workspace-dir",
+			MountPath: "/var/etcd/data",
+		},
+	}
+
+	if etcd.Spec.Backup.Store == nil {
+		return vms
+	}
+
+	storeValues := etcd.Spec.Backup.Store
+
+	if *storeValues.Provider == "GCS" {
+		vms = append(vms, v1.VolumeMount{
+			Name:      "etcd-backup",
+			MountPath: "/root/.gcp/",
+		})
+	}
+
+	return vms
+}
+
+func getCmpctJobVolumes(etcd *druidv1alpha1.Etcd) []v1.Volume {
+	vs := []v1.Volume{
+		{
+			Name: "etcd-workspace-dir",
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: &v1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+
+	if etcd.Spec.Backup.Store == nil {
+		return vs
+	}
+
+	storeValues := etcd.Spec.Backup.Store
+
+	if *storeValues.Provider == "GCS" {
+		vs = append(vs, v1.Volume{
+			Name: "etcd-backup",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: storeValues.SecretRef.Name,
+				},
+			},
+		})
+	}
+
+	return vs
+}
+
+func getCmpctJobEnvVar(etcd *druidv1alpha1.Etcd) []v1.EnvVar {
+	var env []v1.EnvVar
+	if etcd.Spec.Backup.Store == nil {
+		return env
+	}
+
+	storeValues := etcd.Spec.Backup.Store
+
+	env = append(env, getEnvVarFromValues("STORAGE_CONTAINER", *storeValues.Container))
+	//env = append(env, getEnvVarFromFields("POD_NAMESPACE", "metadata.namespace"))
```

Comment:
Needs to be uncommented before merging to add pod namespace as an env var to the compaction job
This is needed by backup sidecar to update the snapshot leases